.github/workflows/ci.yml

```diff
@@ -172,6 +185,28 @@ jobs:
           components: clippy
       - run: cargo +stable clippy --all-features --workspace --all-targets -- -D warnings
 
+  # Try to catch packaging issues that will prevent crate publication.
+  check-publish:
+    name: check (publish)
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@56f84321dbccf38fb67ce29ab63e4754056677e0
+        with:
+          toolchain: nightly
+          components: cargo
+      - name: Check `cargo publish`
+        run: >-
+          cargo +nightly publish
```

Comment:
comment (non-blocking): It occurs to me that this might be slow if there wasn't a previous `cargo +nightly build` on the same machine. I did not check the rest of the workflow definition to see if it was already built with `nightly`, and I trust that you will be keenly aware of any CI time regressions 🙂.

Reply:
It doesn’t do the actual build, just the publication checks. But I might be misunderstanding what you mean by “same machine” here?